src/android/io/branch/BranchSDK.java

```diff
@@ -893,17 +999,38 @@ public void onLinkCreate(String url, BranchError error) {
 
     protected class ShowShareSheetListener implements Branch.BranchLinkShareListener
     {
+
+        private CallbackContext _onShareLinkDialogLaunched;
+        private CallbackContext _onShareLinkDialogDismissed;
+        private CallbackContext _onLinkShareResponse;
+        private CallbackContext _onChannelSelected;
+
+        /**
+         * @constructor
+         *
+         * @param CallbackContext onShareLinkDialogLaunched
+         * @param CallbackContext onShareLinkDialogDismissed
+         * @param CallbackContext onLinkShareResponse
+         * @param CallbackContext onChannelSelected
+         * */
+        public ShowShareSheetListener(CallbackContext onShareLinkDialogLaunched, CallbackContext onShareLinkDialogDismissed, CallbackContext onLinkShareResponse, CallbackContext onChannelSelected) {
+
+            this._onShareLinkDialogDismissed = onShareLinkDialogDismissed;
+            this._onShareLinkDialogLaunched = onShareLinkDialogLaunched;
+            this._onLinkShareResponse = onLinkShareResponse;
+            this._onChannelSelected = onChannelSelected;
+
+        }
+
         @Override
         public void onShareLinkDialogLaunched() {
             Log.d(LCAT, "inside onShareLinkDialogLaunched");
 
-            if (onShareLinkDialogLaunched != null) {
-                PluginResult result = new PluginResult(PluginResult.Status.OK);
+            PluginResult result = new PluginResult(PluginResult.Status.OK);
 
-                result.setKeepCallback(true);
+            result.setKeepCallback(true);
 
-                onShareLinkDialogLaunched.sendPluginResult(result);
-            }
+            this._onShareLinkDialogLaunched.sendPluginResult(result);
```

Comment:
Isn't it possible for this to be null, and then throw a NullPointerException? I think we still want the if ( != null ) check.